string cmd
float64 vol
